ranking/src/experiment_config.rs

```diff
@@ -0,0 +1,77 @@
+use crate::contest_config::ContestSource;
+use crate::compute_ratings::RatingSystem;
+
+#[allow(unused_imports)]
+use crate::{CodeforcesSystem, EloRSystem, TopCoderSystem, TrueSkillSPBSystem};
+
+use serde::{Deserialize};
+use std::path::Path;
+
+#[derive(Deserialize, Debug)]
+pub struct SystemParams {
+    pub method: String,
+    pub params: Vec<f64>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct ExperimentFile {
+    pub max_contests: usize,
+    pub mu_noob: f64,
+    pub sig_noob: f64,
+    pub topk: usize,
+    pub system: SystemParams,
+    pub contest_source: String,
+}
+
+pub struct Experiment {
+    pub max_contests: usize,
+    pub mu_noob: f64,
+    pub sig_noob: f64,
+    pub topk: usize,
+    pub system: Box<dyn RatingSystem>,
+    pub contest_source: ContestSource,
+}
+
+pub fn load_experiment<P: AsRef<Path>>(source: &P) -> Experiment {
+    let params_json =
+        std::fs::read_to_string(source).expect("Failed to read parameters file");
+    let params: ExperimentFile = serde_json::from_str(&params_json).expect("Failed to parse parameters as JSON");
+
+    println!("Loading rating system:\n{:#?}", params);
+    let source = match &params.contest_source[..] {
+        "codeforces" => ContestSource::Codeforces,
+        "reddit" => ContestSource::Reddit,
+        "stackoverflow" => ContestSource::StackOverflow,
+        "synthetic" => ContestSource::Synthetic,
+        _ => ContestSource::NotFound,
+    };
+
+    let system: Box<dyn RatingSystem> = match &params.system.method[..] {
```

Comment:
Hm there are a few ways we might make this conversion safer, but maybe it's best to keep moving for now. We'll just be careful! :+1: